text/3476-expose-fn-type.md

```diff
@@ -0,0 +1,199 @@
+- Feature Name: `expose-fn-type`
+- Start Date: 2023-08-20
+- RFC PR: [rust-lang/rfcs#3476](https://github.com/rust-lang/rfcs/pull/3476)
+- Rust Issue: N/A
+
+# Summary
+[summary]: #summary
+
+This exposes the function type of a function item to the user.
+
+# Motivation
+[motivation]: #motivation
```

Comment:
I'll add more motivation: In Swift, some functions have an associated selector that you can access with [`#selector`](https://developer.apple.com/documentation/swift/using-objective-c-runtime-features-in-swift).

In my crate `objc2`, it would be immensely beautiful (and useful) to be able to do something similar, e.g. access a function's selector using something like `MyClass::my_function::Selector` or `selector(MyClass::my_function)`, instead of having to know the selector name (which might be something completely different than the function name).

Reply:
May I just 1 to 1 copy that sentence? 😅